instance/models/mixins/utilities.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app model mixins - Utilities
+"""
+import logging
+import sys
+
+from django.core.mail import mail_admins
+from django.views.debug import ExceptionReporter
+
+
+logger = logging.getLogger(__name__)
+
+
+class EmailInstanceMixin(object):
+    """
+    An instance class that can send emails
+    """
+    class EmailSubject(object):
+        """
+        Class holding email subject constants
+        """
+        PROVISION_FAILED = u"Instance {instance_name} ({instance_url}) failed to provision"
```

Comment:
Understood, but it adds a little complexity as well:

- Harder to override in subclasses
- Using a class as a namespace feels a bit odd

I feel that simple prefixed constants would make more sense here. To quote the zen of python, "flat is better than nested". But then again the zen of python also says namespaces are great, so what do I know :smile: 

I don't feel too strongly about this, it's your call.